backend/infrahub/actions/schema.py

```diff
@@ -0,0 +1,382 @@
+from infrahub.actions.constants import BranchScope, NodeAction, ValueMatch
+from infrahub.core.constants import AllowOverrideType, BranchSupportType, InfrahubKind
+from infrahub.core.constants import RelationshipCardinality as Cardinality
+from infrahub.core.constants import RelationshipKind as RelKind
+from infrahub.core.schema.attribute_schema import AttributeSchema as Attr
+from infrahub.core.schema.generic_schema import GenericSchema
+from infrahub.core.schema.node_schema import NodeSchema
+from infrahub.core.schema.relationship_schema import (
+    RelationshipSchema as Rel,
+)
+
+core_trigger_rule = GenericSchema(
+    name="TriggerRule",
+    namespace="Core",
+    description="A rule that allows you to define a trigger and map it to an action that runs once the trigger condition is met.",
+    include_in_menu=False,
+    icon="mdi:filter-cog-outline",
+    label="Trigger Rule",
+    default_filter="name__value",
```

Comment:
`default_filter` is deprecated, let's use hfid instead (as well as at other places)